File: ssh-terminal-server/src/domain/vo/ssh.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Clone)]
pub struct SshSessionVO {
    pub id: String,
    pub user_id: String,
    pub name: String,
    pub host: String,
    pub port: u16,
    pub username: String,
    pub group_name: String,
    pub terminal_type: Option<String>,
    pub columns: Option<u16>,
    pub rows: Option<u16>,
    pub auth_method_encrypted: String,
    pub auth_nonce: String,
    pub auth_key_salt: Option<String>,
    pub server_ver: i32,
    pub client_ver: i32,
    pub last_synced_at: Option<i64>,
    pub created_at: i64,
    pub updated_at: i64,
    pub deleted_at: Option<i64>,
}

/// 分页响应
#[derive(Debug, Serialize)]
pub struct PaginatedSshSessions {
    pub data: Vec<SshSessionVO>,
    pub total: u64,
    pub page: u64,
    pub page_size: u64,
}

File: ssh-terminal-web/src/types/ssh.ts
// SSH 会话相关类型
export interface SshSessionVO {
  id: string
  user_id: string
  name: string
  host: string
  port: number
  username: string
  group_name?: string
  terminal_type?: string
  columns?: number
  rows?: number
  auth_method_encrypted: string
  auth_nonce: string
  auth_key_salt?: string
  server_ver: number
  client_ver: number
  last_synced_at?: number
  created_at: number
  updated_at: number
  deleted_at?: number
}

export interface CreateSshSessionRequest {
  name: string
  host: string
  port: number
  username: string
  group_name?: string
  terminal_type?: string
  columns?: number
  rows?: number
  auth_method_encrypted: string
  auth_nonce: string
  auth_key_salt?: string
}

export interface UpdateSshSessionRequest {
  name?: string
  host?: string
  port?: number
  username?: string
  group_name?: string
  terminal_type?: string
  columns?: number
  rows?: number
  auth_method_encrypted?: string
  auth_nonce?: string
  auth_key_salt?: string
}

export interface ListSshSessionsRequest {
  page?: number
  page_size?: number
}

export interface PaginatedSshSessions {
  data: SshSessionVO[]
  total: number
  page: number
  page_size: number
}